lib/Sema/CSDiag.cpp

```diff
@@ -7200,18 +7200,19 @@ bool FailureDiagnosis::visitBindOptionalExpr(BindOptionalExpr *BOE) {
 }
 
 bool FailureDiagnosis::visitIfExpr(IfExpr *IE) {
-  auto typeCheckClauseExpr = [&](Expr *clause) -> Expr * {
-    return typeCheckChildIndependently(clause, Type(), CTP_Unused, TCCOptions(),
+  auto typeCheckClauseExpr = [&](Expr *clause, ContextualTypePurpose convertTypePurpose) -> Expr * {
```

Comment:
`CS.getContextualTypePurpose()` of a `condExpr` evaluates to `CTP_ReturnStmt`, causing the compiler  to emit the old message and exiting
`error: cannot convert return expression of type 'Bool' to return type 'E'`. 

Should I replace `condExpr`'s CTP explicitly to `CTP_Unused`?